app/src/main/java/com/khuttun/notificationnotes/NotificationMgr.java

```diff
@@ -129,4 +163,73 @@ private CharSequence getGroupNotificationLine(NotificationNote note)
         sp.setSpan(new StyleSpan(Typeface.BOLD), 0, note.title.length(), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
         return sp;
     }
+
+    private void displayGroupNotifications(ArrayList<NotificationNote> visibleNotes)
+    {
+        ArrayList<Notification> groupedNotes = createGroupNotifications(visibleNotes);
+        for (int i = 0; i < groupedNotes.size(); i++)
+        {
+            Notification note = groupedNotes.get(i);
+            int id = note.extras.getInt("note.id");
+            if (Globals.LOG)
+            {
+                Log.d(Globals.TAG, "Displaying notification for note with id ".concat(Integer.toString(id)));
+            }
+            this.notificationManager.notify(id, note);
+        }
+    }
+
+    /**
+     * Creates the group of notifications that will be bundled together into one notification.
+     * Used for Android Nougat (7.0) and beyond.
+     */
+    private ArrayList<Notification> createGroupNotifications(ArrayList<NotificationNote> notes)
```

Comment:
If you wouldn't use this function, but rather called createGroupNotification() directly from displayGroupNotifications() while looping through visibleNotes, you wouldn't need the whole "note.id" Bundle, right?